cpp-client-reference/test/test_c99.c

```diff
@@ -0,0 +1,72 @@
+#include "volition.h"
+
+#include <math.h>
+#include <stdio.h>
+
+#define COUNT(arr) (sizeof(arr) / sizeof(arr[0]))
+
+typedef struct {
+    double last_value;
+} simulation_state;
+
+void on_evaluation_request(void* user_data, vl_evaluation_token token, double const* input) {
+    simulation_state* state = (simulation_state*)user_data;
+    double const x1 = input[0];
+    vl_evaluation_result const result = {
+        .output = (double[]){x1 * x1 * state->last_value},
+    };
+    vl_send_evaluation_result(token, &result);
+    state->last_value -= 0.1;
+}
+
+void on_optimization_complete(void* user_data, vl_completion_token token) {
+    printf("optimization complete.\n");
+}
+
+int main() {
+    vl_optimizer const optimizer = vl_create_optimizer("localhost:27016", NULL);
+    if (!optimizer) return -1;
+    vl_optimization_settings const settings = {
+        .target_objective_value = -INFINITY,
+        .concurrent_run_count = 2,
+        .iteration_count = 5,
+    };
+    vl_input const inputs[] = {vl_create_continuous_input("x1", 0.5, 5.0)};
+    vl_client_input const sim_inputs[] = {{.name = "x1", .client_name = "x1"}};
+    vl_client_output const sim_outputs[] = {{.name = "f1", .client_name = "f1", .is_bool = false}};
```

Comment:
what happs if you omit `is_bool = false`? undefined?

Reply:
Members that are not initialized explicitly are [empty-initialized](https://en.cppreference.com/w/c/language/struct_initialization) (zeroed out). So `is_bool` defaults to `false` here. Just wanted to be explicit here in this example.